core/trino-main/src/main/java/io/trino/operator/window/WindowPartition.java

```diff
@@ -83,14 +105,38 @@
 
     private int currentPosition;
 
+    // additional properties for row pattern recognition
+    private final List<MeasureComputation> measures;
+    private final Optional<FrameInfo> commonBaseFrame;
+    private final PatternRecognitionRelation.RowsPerMatch rowsPerMatch;
+    private final Optional<LogicalIndexNavigation> skipToNavigation;
+    private final SkipTo.Position skipToPosition;
+    private final boolean initial;
+    private final Optional<Machine> machine;
+    private final Evaluation[] labelEvaluations;
```

Comment:
Why not `List<Evaluation>`? It's safer, especially if the list can be immutable.